main.go

```diff
@@ -1,288 +1,134 @@
-	"encoding/json"
+	// "log"
 	"fmt"
-	"io/ioutil"
-	"net/http"
-	"net/http/httputil"
 	"os"
-	"path"
-	"strings"
 
-	"github.com/drone/drone-go/drone"
-	"github.com/drone/drone-go/plugin"
+	"github.com/arcolife/jfrog-client-go/utils/log"
+	"github.com/joho/godotenv"
+	"github.com/sirupsen/logrus"
+	"github.com/urfave/cli/v2"
 )
 
-const defaultHost = "https://api.bintray.com"
-
-type Bintray struct {
-	Username  string     `json:"username"`
-	APIKey    string     `json:"api_key"`
-	Branch    string     `json:"branch"`
-	Host      string     `json:"host"`
-	Debug     bool       `json:"debug"`
-	Insecure  bool       `json:"insecure"`
-	Artifacts []Artifact `json:"artifacts"`
-}
-
-type Artifact struct {
-	File       string      `json:"file"`
-	Type       string      `json:"type"`
-	Owner      string      `json:"owner"`
-	Repository string      `json:"repository"`
-	Artifact   string      `json:"package"`
-	Target     string      `json:"target"`
-	Distr      string      `json:"distr,omitempty"`
-	Component  string      `json:"component,omitempty"`
-	Arch       []string    `json:"arch,omitempty"`
-	Publish    bool        `json:"publish,omitempty"`
-	Override   bool        `json:"override,omitempty"`
-	Versioni   interface{} `json:"version"`
-	Version    string
-}
-
 var (
-	version string
-	bintray Bintray
+	Version = "unknown"
+	file    *os.File
 )
 
-type MessageText struct {
-	Message string `json:"message"`
-}
-
 func main() {
-	fmt.Printf("Drone Bintray Plugin built from %s\n", version)
-
-	var workspace = drone.Workspace{}
-
-	plugin.Param("workspace", &workspace)
-	plugin.Param("vargs", &bintray)
-	if err := plugin.Parse(); err != nil {
-		fmt.Printf("ERROR Can't parse yaml config: %s", err.Error())
-		os.Exit(1)
-	}
-
-	if bintray.Host == "" {
-		bintray.Host = defaultHost
-	}
-
-	if bintray.Debug {
-		saveApikey := bintray.APIKey
-		bintray.APIKey = "******"
-		fmt.Printf("DEBUG plugin input:\n%#v\n%#v\n", workspace, bintray)
-		bintray.APIKey = saveApikey
-	}
-	if len(bintray.Branch) == 0 || bintray.Branch == "master" {
-		fmt.Printf("\nPublishing %d artifacts to Bintray for user %s\n", len(bintray.Artifacts), bintray.Username)
-	} else {
-		fmt.Printf("\nPublishing %d artifacts on branch %s to Bintray for user %s\n", len(bintray.Artifacts), bintray.Branch, bintray.Username)
-	}
-	for i, artifact := range bintray.Artifacts {
-		artifact.Version = fmt.Sprintf("%v", artifact.Versioni)
-		fmt.Printf("\nUploading file %d %s to %s\n", i+1,
-			artifact.File, artifact.getEndpoint())
-		artifact.Upload(workspace.Path)
+	if filename, found := os.LookupEnv("PLUGIN_ENV_FILE"); found {
+		_ = godotenv.Load(filename)
+	}
+
+	app := cli.NewApp()
+	app.Name = "bintray-uploader plugin"
+	app.Usage = "bintray-uploader plugin"
+	app.Action = run
+	app.Version = Version
+	app.Flags = []cli.Flag{
+		&cli.IntFlag{
+			Name:    "bintray.threads,t",
+			Usage:   "bintray threads",
+			EnvVars: []string{"PLUGIN_BINTRAY_THREADS", "BINTRAY_THREADS"},
```

Comment:
As this is the bintray plugin please name them PLUGIN_ instead of PLUGIN_BINTRAY_

Reply:
noted